### PR TITLE
[FLINK-10274][flink-core]Add `env.pid.dir` configuration to CoreOptions

### PR DESCRIPTION
**Pull request by bytesmithing**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

This pull request Ensure that the documentation does not confuse developers. Because I cannot find the `env.pid.dir` configuration in the documentation, but the conf will used in the `conf.sh`.  And reminde the developer who want to override it where to modify.

## Brief change log

- add `env.pid.dir` configuration to Configuration page of Documentation.

## Verifying this change

This change is already covered by existing tests, such as Flink-docs test

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):  no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers:  no
  - The runtime per-record code paths (performance sensitive): don't know
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: don't know
  - The S3 file system connector: don't know

## Documentation

  - Does this pull request introduce a new feature?  no

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java [hunk 1]
```diff
@@ -218,6 +218,17 @@
 		.withDescription("Defines the directory where the Flink logs are saved. It has to be an absolute path." +
 			" (Defaults to the log directory under Flink’s home)");
 
+	/**
+	 * This options is here only for documentation generation, it is only
+	 * evaluated in the shell scripts.
+	 */
+	public static final ConfigOption<String> FLINK_APP_PID_DIR = ConfigOptions
+		.key("env.pid.dir")
+		.noDefaultValue()
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java:227]: the default is "/tmp", which we should document

File: flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java [hunk 2]
```diff
@@ -218,6 +218,17 @@
 		.withDescription("Defines the directory where the Flink logs are saved. It has to be an absolute path." +
 			" (Defaults to the log directory under Flink’s home)");
 
+	/**
+	 * This options is here only for documentation generation, it is only
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java:222]: It would be good to explicitly link to `config.sh#KEY_ENV_PID_DIR `and `config.sh#DEFAULT_ENV_PID_DIR`.

File: flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java [hunk 3]
```diff
@@ -218,6 +218,17 @@
...
+	 * evaluated in the shell scripts.
+	 */
+	public static final ConfigOption<String> FLINK_APP_PID_DIR = ConfigOptions
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java:225]:
```suggestion
	public static final ConfigOption<String> FLINK_PID_DIR = ConfigOptions
```

File: flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java [hunk 4]
```diff
@@ -218,6 +218,17 @@
...
+		.key("env.pid.dir")
+		.noDefaultValue()
+		.withDescription(
+			"Defines the directory where the flink-<host>-<process>.pid files are saved."
+				+ "(Defaults to the system tmp directory)");
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java:230]:
```suggestion
			"Defines the directory where the flink-<host>-<process>.pid files are saved.");
```

**Review by zentol — CHANGES_REQUESTED** [review]
You need to regenerate the documentation; you can find instructions to do so [here](https://github.com/apache/flink/tree/master/flink-docs#configuration-documentation).

File: flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java (hunk 2 above)
Comment (bytesmithing) [reply flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java:222]: I cannot link the config.sh file  in the java document with {@link}. Can you help me ?

**Comment by bytesmithing** [comment]
Thanks god, I finished the project building and regenerated the doc. Please take a look. @zentol

File: flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java (hunk 2 above)
Comment (zentol) [reply flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java:222]: You can just refer to the files as "bin/config.sh"; that's sufficient.

File: docs/_includes/generated/yarn_config_configuration.html [hunk 5]
```diff
@@ -128,6 +122,12 @@
             <td>List&lt;String&gt;</td>
             <td>A semicolon-separated list of provided lib directories. They should be pre-uploaded and world-readable. Flink will use them to exclude the local Flink jars(e.g. flink-dist, lib/, plugins/)uploading to accelerate the job submission process. Also YARN will cache them on the nodes so that they doesn't need to be downloaded every time for each application. An example could be hdfs://$namenode_address/path/of/flink/lib</td>
         </tr>
+        <tr>
+            <td><h5>yarn.security.kerberos.additionalFileSystems</h5></td>
+            <td style="word-wrap: break-word;">(none)</td>
```
Comment (zentol) [inline docs/_includes/generated/yarn_config_configuration.html:127]: can you put this change into a separate commit labeled "[hotfix][docs] Fix order"? Looks like someone modified the file manually once again...

File: flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java (hunk 2 above)
Comment (bytesmithing) [reply flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java]: Thanks a lot. Please take a look.
